csd/kyuubi-csd-src/scripts/control.sh

```diff
@@ -0,0 +1,134 @@
+    echo rootLogger.appenderRef.DRFA.level = ${logevel}                  >> conf/log4j2.properties
+    echo property.log.dir = ${KYUUBI_LOG_DIR}                                         >> conf/log4j2.properties
+    echo property.log.file = ${logfile}                                       >> conf/log4j2.properties
+    echo appenders = console, DRFA                                             >> conf/log4j2.properties
+    echo property.max.log.file.size = ${maxsize}                     >> conf/log4j2.properties
+    echo property.max.log.file.backup.index = ${maxindex}     >> conf/log4j2.properties
+    echo appender.DRFA.type = RollingRandomAccessFile                          >> conf/log4j2.properties
+    echo appender.DRFA.name = DRFA                                             >> conf/log4j2.properties
+    echo appender.DRFA.fileName = \${log.dir}/\${log.file}                       >> conf/log4j2.properties
+    echo appender.DRFA.filePattern = \${log.dir}/\${log.file}.%d{yyyy-MM-dd}     >> conf/log4j2.properties
+    echo appender.DRFA.layout.type = PatternLayout                             >> conf/log4j2.properties
+    echo appender.DRFA.layout.pattern = %d{DEFAULT} %-5p %c: [%t]: %m%n        >> conf/log4j2.properties
+    echo appender.DRFA.policies.type = Policies                                >> conf/log4j2.properties
+    echo appender.DRFA.policies.size.type = SizeBasedTriggeringPolicy          >> conf/log4j2.properties
+    echo appender.DRFA.policies.size.size = \${max.log.file.size}               >> conf/log4j2.properties
+    echo appender.DRFA.strategy.type = DefaultRolloverStrategy                 >> conf/log4j2.properties
+    echo appender.DRFA.strategy.max = \${max.log.file.backup.index}             >> conf/log4j2.properties
+    echo appender.console.type = Console                                       >> conf/log4j2.properties
+    echo appender.console.name = console                             >> conf/log4j2.properties
+    echo appender.console.target = SYSTEM_ERR                        >> conf/log4j2.properties
+    echo appender.console.layout.type = PatternLayout                >> conf/log4j2.properties
+    echo appender.console.layout.pattern = %d{yy/MM/dd HH:mm:ss} %-5p %c{2}: [%t]: %m%n       >> conf/log4j2.properties
+    CLASS="org.apache.kyuubi.server.KyuubiServer"
+    RUNNER="${JAVA_HOME}/bin/java"
+    if [[ -z "$KYUUBI_JAR_DIR" ]]; then
+  	KYUUBI_JAR_DIR="$KYUUBI_HOME/jars"
+    	if [[ ! -d ${KYUUBI_JAR_DIR} ]]; then
+  		echo -e "\nCandidate Kyuubi lib $KYUUBI_JAR_DIR doesn't exist, searching development environment..."
+    		KYUUBI_JAR_DIR="$KYUUBI_HOME/kyuubi-assembly/target/scala-${KYUUBI_SCALA_VERSION}/jars"
+  	fi
+    fi
+    if [[ -z ${YARN_CONF_DIR} ]]; then
+  	KYUUBI_CLASSPATH="${KYUUBI_JAR_DIR}/*:${KYUUBI_CONF_DIR}:${HADOOP_CONF_DIR}"
+    else
+  	KYUUBI_CLASSPATH="${KYUUBI_JAR_DIR}/*:${KYUUBI_CONF_DIR}:${HADOOP_CONF_DIR}:${YARN_CONF_DIR}"
```

Comment:
the whole scripts is kind of hacky, `bin/kyuubi run` can run KyuubiServer in foreground, have you tried this way?

Reply:
No, at that time, the test used bin/kyuubi run to exit the process directly, and other processes were executing, for fear that the supervisor would not work, so this method was not used; Let me modify the test and use bin/kyuubi run

Reply:
There are still some problems with using bin/kyuubi script, kyuubi_ HOME, kyuubi-env.sh and other parameters in the script are set. cm is automatically generated according to the page options, especially the configuration file under conf. If the conf generated by cm is placed under kyuubi/conf in parcels, there are still some permission problems to debug. Please see if you need to do this